[rom] Fix pre-signed tests

cfrantz opened the pull request:
1. Convert the e2e_rom_self_hash test to a pre-signed binary test.
2. Update the hashes in the ROM self hash test now that the real HSM keys are committed to the codebase.
3. Sign the empty_test and e2e_rom_self_hash test with the real `earlgrey_a0_test_0` key.

alphan commented on sw/device/silicon_creator/rom/e2e/BUILD:
```diff
@@ -3920,36 +3920,46 @@ test_suite(
     name = "presigned_fpga_binary",
     cw310 = cw310_params(
-        bitstream = ":bitstream_sigverify_spx_prod_enabled_zero",
+        bitstream = "//hw/bitstream:rom_with_real_keys_otp_test_unlocked0",
```
note that this test was signed with both RSA and SPHINCS+ keys. This change seems to be for RSA-only signature verification. is the RSA + SPHINCS+ case tracked somewhere?

cfrantz replied:
See #18860.

Because we don't have a secure signing solution for SPX keys (HSM or personal token), I'm nervous about signing this and then having to make a last minute change to the ROM.

cfrantz replied:
I had the self hash test on the brain - we can probably sign the `empty_test` with an SPX key and test that.

alphan replied:
sg, thanks!

alphan commented on sw/device/silicon_creator/rom/e2e/BUILD:
```diff
@@ -3920,36 +3920,46 @@ test_suite(
 
-# TODO(#18686): update to be pres-signed
-opentitan_functest(
+opentitan_flash_binary(
```
this target is just for building binary to be signed, right? This is a minor point but wouldn't it be nice to be able to run this test with fake keys before signing the binary?

cfrantz replied:
We'd need another copy of this test: the golden hash encoded in the test source code is for `rom_with_real_keys`.

timothytrippel commented on sw/device/silicon_creator/rom/e2e/presigned_images/BUILD:
```diff
@@ -13,3 +13,53 @@ filegroup(
+    srcs = ["rom_e2e_self_hash_sim_dv.signed.bin"],
+)
+
+alias(
```
just dropping a note here: this alias is only needed b/c the `opentitan_functest` rule expects to find a target with the name `*_<device>_bin` when `signed = False` and `*_<device>_signed_bin` when `signed = True`, but if the above filegroups were named accordingly, then the aliases would not be needed I don't think (we should probably drop a comment explaining this so these label names are not updated in the future.

alphan replied:
yeah, good point. That's why I added those when I first wrote this file but forgot to add a comment.

timothytrippel replied:
@cfrantz tried this and it did not work, we will investigate in a follow up.

cfrantz commented:
Per this morning's discussion, I've adjusted the sha256 hashes to omit `chip_info` and I've added code to print out the values of `chip_info`.

I've created TODO #18868 to add checks for the chip_info values we expect to see in the released ROM binary.
